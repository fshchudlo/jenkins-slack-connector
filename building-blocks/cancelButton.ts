import { ActionKeys } from "../actions";
import { button } from "./button";

export function cancelButton(text: string = "Cancel", action_id: string = ActionKeys.DISPLAY_CANCELLATION_MESSAGE, value: string = "Cancel", style: string = "danger") {
    return button(text, action_id, value, style);
}