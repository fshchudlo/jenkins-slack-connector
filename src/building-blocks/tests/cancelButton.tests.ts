import { cancelButton } from "../cancelButton";

describe("cancelButton", () => {
    it("should create a cancel button with the correct properties", () => {
        const control = cancelButton();
        expect(control).toMatchSnapshot();
    });

    it("should use the provided values if provided", () => {
        const control = cancelButton("Don't do it", "cancel_action", "no", "warning");
        expect(control).toMatchSnapshot();
    });
});
