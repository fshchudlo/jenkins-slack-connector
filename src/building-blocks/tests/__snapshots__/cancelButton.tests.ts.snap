// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`cancelButton should create a cancel button with the correct properties 1`] = `
{
  "action_id": "display_cancellation_message",
  "style": "danger",
  "text": {
    "text": "Cancel",
    "type": "plain_text",
  },
  "type": "button",
  "value": "Cancel",
}
`;

exports[`cancelButton should use the provided values if provided 1`] = `
{
  "action_id": "cancel_action",
  "style": "warning",
  "text": {
    "text": "Don't do it",
    "type": "plain_text",
  },
  "type": "button",
  "value": "no",
}
`;
